File: КОМПОНЕНТЫ/OPartner/Examples/CBuildr5/exppt1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "ExPpt1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "OpPower"
#pragma link "OpPpt2k"
#pragma link "OpShared"
#pragma resource "*.dfm"
TForm1 *Form1;
TOpPresentation* Pres;
Opppt2k::_di__Presentation IPres;

//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
  cbxLayout->ItemIndex = 0;
  cbxTransitionEffect->ItemIndex = 0;
  cbxTransitionSpeed->ItemIndex = 0;
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnOpenPresentationClick(TObject *Sender)
{
  if (OpenDialog1->Execute()) {
    OpPowerPoint1->Connected = true;
    Pres = OpPowerPoint1->OpenPresentation(OpenDialog1->FileName);
    IPres = Pres->AsPresentation;
    IPres->SlideShowSettings->AdvanceMode = ppSlideShowUseSlideTimings;
  }
}
//---------------------------------------------------------------------------

void __fastcall TForm1::FormClose(TObject *Sender, TCloseAction &Action)
{
  btnClosePresentationClick(0);
  OpPowerPoint1->Quit();
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnRunSlideShowClick(TObject *Sender)
{
  if (Pres->Slides->Count > 0)
    for (int i = 0; i < Pres->Slides->Count; i++) {
      TOpSlide* Sld = Pres->Slides->Items[i];
      Sld->Layout = TOpPpSlideLayout(cbxLayout->ItemIndex);
      Sld->TransitionEffect->EntryEffect =
        TOpPpEntryEffect(cbxTransitionEffect->ItemIndex);
      Sld->TransitionEffect->Speed =
        TOpPpTransitionSpeed(cbxTransitionSpeed->ItemIndex);
      Sld->TransitionEffect->AdvanceOnClick = chkAdvanceOnClick->Checked;
      Sld->TransitionEffect->AdvanceOnTime  = chkAdvanceOnTime->Checked;
      if (chkAdvanceOnTime->Checked)
        Sld->TransitionEffect->AdvanceTime = StrToIntDef(edtAdvanceTime->Text, 10);
      Pres->RunSlideShow();
    }
}
//---------------------------------------------------------------------------

void __fastcall TForm1::btnClosePresentationClick(TObject *Sender)
{
  if (Pres)
    delete Pres;
  Pres = 0;
  OpPowerPoint1->Connected = false;
}
//---------------------------------------------------------------------------

File: КОМПОНЕНТЫ/OPartner/Examples/Delphi/exppt1.pas
unit ExPpt1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, OpShared, OpPpt2k, OpPower, ExtCtrls;

type
  TForm1 = class(TForm)
    btnOpenPresentation: TButton;
    btnRunSlideShow: TButton;
    OpPowerPoint1: TOpPowerPoint;
    OpenDialog1: TOpenDialog;
    btnClosePresentation: TButton;
    gbOptions: TGroupBox;
    Label1: TLabel;
    cbxTransitionEffect: TComboBox;
    Label2: TLabel;
    cbxLayout: TComboBox;
    edtAdvanceTime: TEdit;
    Label3: TLabel;
    cbxTransitionSpeed: TComboBox;
    Label4: TLabel;
    chkAdvanceOnClick: TCheckBox;
    chkAdvanceOnTime: TCheckBox;
    procedure btnOpenPresentationClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnRunSlideShowClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure btnClosePresentationClick(Sender: TObject);
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

var
  Pres : TOpPresentation;
  IPres : OpPpt2K.Presentation;

procedure TForm1.FormCreate(Sender: TObject);
begin
  cbxLayout.ItemIndex := 0;
  cbxTransitionEffect.ItemIndex := 0;
  cbxTransitionSpeed.ItemIndex := 0;
end;

procedure TForm1.btnOpenPresentationClick(Sender: TObject);
begin
  if OpenDialog1.Execute then begin
    OpPowerPoint1.Connected := True;
    Pres := OpPowerPoint1.OpenPresentation(OpenDialog1.FileName);
    IPres := Pres.AsPresentation;
    IPres.SlideShowSettings.AdvanceMode := ppSlideShowUseSlideTimings;
  end;
end;

procedure TForm1.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  btnClosePresentationClick(nil);
  OpPowerPoint1.Quit;
end;

procedure TForm1.btnRunSlideShowClick(Sender: TObject);
var
  i : Integer;
begin
  { slide options }
  if (Pres.Slides.Count > 0) then
    for i := 0 to Pred(Pres.Slides.Count) do
      with Pres.Slides[i] do begin
        Layout := TOpPpSlideLayout(cbxLayout.ItemIndex);
        with TransitionEffect do begin
          EntryEffect := TOpPpEntryEffect(cbxTransitionEffect.ItemIndex);
          Speed := TOpPpTransitionSpeed(cbxTransitionSpeed.ItemIndex);
          AdvanceOnClick := chkAdvanceOnClick.Checked;
          AdvanceOnTime  := chkAdvanceOnTime.Checked;
          if AdvanceOnTime then
            AdvanceTime := StrToIntDef(edtAdvanceTime.Text, 10);
        end;
      end;

  Pres.RunSlideShow;
end;

procedure TForm1.btnClosePresentationClick(Sender: TObject);
begin
  if Assigned(Pres) then
    Pres.Free;
  Pres := nil;
  OpPowerPoint1.Connected := False;
end;

end.
